features/support/cucumber_rails_helper.rb

```diff
@@ -108,11 +104,15 @@ def add_conditional_gems(options)
     else
       add_gem 'cucumber-rails', group: :test, require: false, path: File.expand_path('.').to_s
     end
+  end
 
-    if rails_6_0_or_higher?
+  def add_rails_specific_gems
+    if rails_equal_or_higher_than?('6.0')
       add_gem 'sqlite3', '~> 1.4'
+      add_gem 'selenium-webdriver', '~> 4', group: :test
```

Comment:
The `selenium-webdriver` uses the `rexml` library. Downstream projects only need `rexml` if they are also using the optional Selenium framework.

The `selenium-webdriver` maintainers have added an explicit runtime dependency in version 4. This resolves the test errors on Ruby 3.